### PR TITLE
Feat/floor staff

### PR DESCRIPTION
**Pull request by 21434809**
added the element part of the visitor pattern in the staff file there is a Manager and a Waiter that that are both concrete elements with an accept function  that takes in a  a customer state

### CODE REVIEW

File: src/staff/Waiter.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+     * @brief Default constructor for Waiter.
+     */
+    Waiter();
+
+    /**
+     * @brief Checks the kitchen.
+     *
+     * This function is used to check the kitchen for orders.
+     */
+    void checkKitcken();
```
Comment (Vafdaf12) [inline src/staff/Waiter.h]: Kitcken???? 👀

File: src/staff/Manager.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "FloorStaff.h"
+#include "customer/CustomerState.h"
+#include <iostream>
+using namespace std;
```
Comment (Vafdaf12) [inline src/staff/Manager.h]: No `using namespace std` (as seen in the Style Guide)

File: src/staff/Manager.h
Comment (Vafdaf12) [inline src/staff/Manager.h:1]: Please include some functions for customer satisfaction (e.g. lodging a complaint, giving a rating, etc.)

Comment (Vafdaf12) [inline src/staff/Manager.h:1]: Would the Manager then also be capable of taking orders from customers and providing customers with meals?

File: src/staff/Waiter.h (hunk 1 above)
Comment (Vafdaf12) [reply src/staff/Waiter.h]: hi

**Review by Vafdaf12 — CHANGES_REQUESTED** [review]

File: src/staff/Manager.h
Comment (21434809) [reply src/staff/Manager.h:1]: no thats wrong

File: src/staff/Waiter.h (hunk 1 above)
Comment (21434809) [reply src/staff/Waiter.h]: kitchen

File: src/staff/Manager.h (hunk 2 above)
Comment (21434809) [reply src/staff/Manager.h]: its aids

File: src/staff/Manager.h
Comment (21434809) [reply src/staff/Manager.h:1]: is needed

**Review by Vafdaf12 — APPROVED** [review]
Happy days :)

**Pull request merged**